src/mediapipe_internal/mediapipegraphexecutor.hpp

```diff
@@ -51,23 +93,209 @@ class MediapipeGraphExecutor {
+        // can be still processing those. Closing packet sources triggers Calculator::Close() on nodes that do not expect
+        // new packets
+        MP_RETURN_ON_FAIL(graph.WaitUntilIdle(), "graph wait until idle", StatusCode::MEDIAPIPE_EXECUTION_ERROR);
+        MP_RETURN_ON_FAIL(graph.CloseAllPacketSources(), "graph close all packet sources", StatusCode::MEDIAPIPE_GRAPH_CLOSE_INPUT_STREAM_ERROR);
+        for (auto& [outputStreamName, poller] : outputPollers) {
+            size_t receivedOutputs = 0;
+            SPDLOG_DEBUG("Will wait for output stream: {} packet", outputStreamName);
+            if (poller.Next(&packet)) {
+                SPDLOG_DEBUG("Received packet from output stream: {}", outputStreamName);
+                try {
+                    OVMS_RETURN_ON_FAIL(
+                        onPacketReadySerializeImpl(
+                            request->id(),
+                            this->name,
+                            this->version,
+                            outputStreamName,
+                            this->outputTypes.at(outputStreamName),
+                            packet,
+                            *response));
+                } catch (...) {
+                    return Status(StatusCode::MEDIAPIPE_EXECUTION_ERROR, "Exception during packet serialization");
+                }
+                outputPollersWithReceivedPacket.insert(outputStreamName);
+                ++receivedOutputs;
+            }
+            SPDLOG_TRACE("Received all: {} packets for: {}", receivedOutputs, outputStreamName);
+        }
+        MP_RETURN_ON_FAIL(graph.WaitUntilDone(), "grap wait until done", StatusCode::MEDIAPIPE_EXECUTION_ERROR);
+        if (outputPollers.size() != outputPollersWithReceivedPacket.size()) {
+            SPDLOG_DEBUG("Mediapipe failed to execute. Failed to receive all output packets");
+            return Status(StatusCode::MEDIAPIPE_EXECUTION_ERROR, "Unknown error during mediapipe execution");
+        }
+        SPDLOG_DEBUG("Received all output stream packets for graph: {}", this->name);
+        return StatusCode::OK;
+    }
+
+    template <typename RequestType, typename ResponseType>
+    Status inferStream(const RequestType& req, ResponseType& res) {
+        SPDLOG_DEBUG("Start MediapipeGraphExecutor::inferEx mediapipe graph: {} execution", this->name);
+        std::mutex sendMutex;
+        try {
+            // Init
+            ::mediapipe::CalculatorGraph graph;
+            MP_RETURN_ON_FAIL(graph.Initialize(this->config), "graph initialization", StatusCode::MEDIAPIPE_GRAPH_INITIALIZATION_ERROR);
+
+            // Installing observers
+            for (const auto& outputName : this->outputNames) {
+                MP_RETURN_ON_FAIL(graph.ObserveOutputStream(outputName, [&res, &sendMutex, &outputName, this](const ::mediapipe::Packet& packet) -> absl::Status {
+                    try {
+                        // API
```

Comment:
What's up with these `// API` comments in this file?

Reply:
Since this is template definition, I wanted to define places where custom implementation will be required (currently only kfs_graph_executor_impl). If that is too much of information left in the code, let me know.

Reply:
I think it's either too much or too little. If we want to give such information, I'd prefer it to be more explicit and verbose or not included it at all.

Reply:
It's a suggestion that it is part of the code one needs to implement when enabling specific frontend. However if it's too little, full detailed explanation of this call is in header kfs_graph_executor_impl.hpp

Reply:
Removed